moz_sql_parser/sql_parser.py

```diff
@@ -41,281 +46,204 @@ def nothing(*args):
+INNERJOIN = Keyword("inner join", caseless=True).setName("inner join").addParseAction(an.InnerJoinKeyword.from_tokens).setDebugActions(*debug)
+IS = Keyword("is", caseless=True).setName("is").addParseAction(an.IsKeyword.from_tokens).setDebugActions(*debug)
+JOIN = Keyword("join", caseless=True).setName("join").addParseAction(an.JoinKeyword.from_tokens).setDebugActions(*debug)
+LEFTJOIN = Keyword("left join", caseless=True).setName("left join").addParseAction(an.LeftJoinKeyword.from_tokens).setDebugActions(*debug)
+LEFTOUTERJOIN = Keyword("left outer join", caseless=True).setName("left outer join").addParseAction(an.LeftOuterJoinKeyword.from_tokens).setDebugActions(*debug)
+LIMIT = Keyword("limit", caseless=True).setName("limit").addParseAction(an.LimitKeyword.from_tokens).setDebugActions(*debug)
+OFFSET = Keyword("offset", caseless=True).setName("offset").addParseAction(an.OffsetKeyword.from_tokens).setDebugActions(*debug)
+LIKE = Keyword("like", caseless=True).setName("like").addParseAction(an.LikeKeyword.from_tokens).setDebugActions(*debug)
+NOTLIKE = Keyword("not like", caseless=True).setName("not like").addParseAction(an.NotLikeKeyword.from_tokens).setDebugActions(*debug)
+ON = Keyword("on", caseless=True).setName("on").addParseAction(an.OnKeyword.from_tokens).setDebugActions(*debug)
+USING = Keyword("using", caseless=True).setName("using").addParseAction(an.UsingKeyword.from_tokens).setDebugActions(*debug)
+OR = Keyword("or", caseless=True).setName("or").addParseAction(an.OrKeyword.from_tokens).setDebugActions(*debug)
+ORDERBY = Keyword("order by", caseless=True).setName("order by").addParseAction(an.OrderByKeyword.from_tokens).setDebugActions(*debug)
+RIGHTJOIN = Keyword("right join", caseless=True).setName("right join").addParseAction(an.RightJoinKeyword.from_tokens).setDebugActions(*debug)
+RIGHTOUTERJOIN = Keyword("right outer join", caseless=True).setName("right outer join").addParseAction(an.RightOuterJoinKeyword.from_tokens).setDebugActions(*debug)
+SELECT = Keyword("select", caseless=True).setName("select").addParseAction(an.SelectKeyword.from_tokens).setDebugActions(*debug)
+THEN = Keyword("then", caseless=True).setName("then").addParseAction(an.ThenKeyword.from_tokens).setDebugActions(*debug)
+UNION = Keyword("union", caseless=True).setName("union").addParseAction(an.UnionKeyword.from_tokens).setDebugActions(*debug)
+UNIONALL = Keyword("union all", caseless=True).setName("union all").addParseAction(an.UnionAllKeyword.from_tokens).setDebugActions(*debug)
+WHEN = Keyword("when", caseless=True).setName("when").addParseAction(an.WhenKeyword.from_tokens).setDebugActions(*debug)
+WHERE = Keyword("where", caseless=True).setName("where").addParseAction(an.WhereKeyword.from_tokens).setDebugActions(*debug)
+WITH = Keyword("with", caseless=True).setName("with").addParseAction(an.WithKeyword.from_tokens).setDebugActions(*debug)
```

Comment:
On one hand I am happy you did this so my linter does not complain about missing variables.  On the other hand it is a wall of redundant code that is difficult to verify.  May I suggest the original `keywords`  be CamelCase, and then perform:

```python
for k in keywords:
    name = k.upper().replace(" ", "")
    locs[name] = value = Keyword(k, caseless=True)
        .setName(k.lower())
        .addParseAction(getattr(an, k+"Keyword").from_tokens)
        .setDebugActions(*debug)
    reserved.append(value)
```

Reply:
Right, that would work, but as you said PyCharm and other IDEs really do not like it. I'm not happy either way really. I could even add the actual keyword to the ASTNode as a class variable.

Reply:
We can consider putting the keyword generation in its own file, and them imported here.  This way the linters stop complaining about the main code.